### ISSUE TITLE
Some times it takes a lot of time to Leave online servers

### ISSUE THREAD

**Issue by M0REKZ** [efroemling/ballistica#707]
### Description
Trying to leave some online servers will make the game get stuck for a while, you must wait a lot of time until it actually returns to the server list.

### Steps to reproduce
1. Launch BombSquad
2. Go to Online Party (Public Server List)
3. Join and Leave Random Servers until bug happens
4. You will get stuck trying to leave a server at some point, this can happen even on the first try.

### Expected behavior

Leave the server should not take that long nor forever

### Machine
**Platform**: macOS 14.5
**BombSquad version**: Mac 1.7.35 from https://ballistica.net/downloads 2024-05-16 | 21879

You may specify BombSquad version you're running or refer to the latest commit.

### Screenshots

https://github.com/efroemling/ballistica/assets/60852359/3d302cd2-ef37-4e3f-9569-58f81b6d1fbf

**Comment by vishal332008** [comment]
Should be the server delay to remove you from the server. No biggie.

**Comment by M0REKZ** [comment]
> Should be the server delay to remove you from the server. No biggie.

It's still kinda annoying, the first times i had think that the Game was crashing.

**Comment by MatiasMGz** [comment]
Yeah, it has happened since, maybe, the first time i played on public servers. It is annoying and i would like this to be fixed.

**Comment by EraOSBeta** [comment]
Would be nice if the game could just handle disconnections in a background thread so that players don't have to deal with this.

**Comment by MatiasMGz** [comment]
Yeah.

**Comment by HeyErfan** [comment]
When the internet is disconnected or becomes very weak, exactly this happens. You cannot leave the server (until the internet reaches a relatively good level and the server can kick you out)

**Comment by HypeStratex** [comment]
Agreed, this should be solved, the solution of @EraOSBeta sounds good.
